### PR TITLE
Updated server.go to include metrics for pending dial requests

### PR DESCRIPTION
**Pull request by jdnurme**

### CODE REVIEW

File: pkg/server/metrics/metrics.go [hunk 1]
```diff
@@ -134,3 +154,13 @@ func (a *ServerMetrics) ConnectionDec(serviceMethod string) {
 func (a *ServerMetrics) SetBackendCount(count int) {
 	a.backend.WithLabelValues().Set(float64(count))
 }
+
+// PendingDialInc increments a pending agent dial.
```
Comment (cheftako) [inline pkg/server/metrics/metrics.go:158]: I think its safer to set the count then trust to always get the increment/decrement called.

File: pkg/server/server.go [hunk 2]
```diff
@@ -418,6 +418,7 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 				// The Dial is failing; no reason to keep this goroutine.
 				return
 			}
+			metrics.Metrics.PendingDialInc(metrics.ServeRecvFrontend)
```
Comment (cheftako) [inline pkg/server/server.go:421]: Can we do something like ```metrics.Metrics.PendingDialSet(len(s.PendingDial))```? That way if someone changes PendingDial without calling the metric inc/dec your haven't leaked a number.

File: pkg/server/metrics/metrics.go (hunk 1 above)
Comment (jdnurme) [reply pkg/server/metrics/metrics.go]: ack

**Pull request merged**